glide-core/telemetry/src/open_telemetry.rs

```diff
@@ -478,6 +478,99 @@ static OTEL: OnceCell<RwLock<GlideOpenTelemetry>> = OnceCell::new();
 
 /// Our interface to OpenTelemetry
 impl GlideOpenTelemetry {
+    /// Validate if a span pointer is valid and safe to use
+    ///
+    /// # Arguments
+    /// * `span_ptr` - The u64 span pointer to validate
+    ///
+    /// # Returns
+    /// * `true` - If the span pointer is valid (non-zero and within reasonable bounds)
+    /// * `false` - If the span pointer is invalid (zero, out of bounds, or potentially corrupted)
+    ///
+    /// # Safety
+    /// This function performs basic validation but cannot guarantee the pointer points to valid memory.
+    /// It only checks for obvious invalid values like null pointers and unreasonable addresses.
+    pub fn is_span_pointer_valid(span_ptr: u64) -> bool {
```

Comment:
This needs to be marked unsafe, because the caller can pass in any arbitrary `u64` value, not necessarily a valid span pointer.